### PR TITLE
Deserialize stream

### PR DESCRIPTION
**Pull request by abhipsaMisra**

### CODE REVIEW

File: iothub/device/src/Transport/Http/HttpClientHelper.cs [hunk 1]
```diff
@@ -236,10 +236,9 @@ private static async Task<T> ReadAsAsync<T>(HttpContent content, CancellationTok
         {
             token.ThrowIfCancellationRequested();
 
+            var jsonSerializerOptions = new JsonSerializerOptions();
```
Comment (drwill-ms) [inline iothub/device/src/Transport/Http/HttpClientHelper.cs:239]: Do we need options?

**Review by drwill-ms — APPROVED** [review]

File: iothub/device/src/Transport/Http/HttpClientHelper.cs (hunk 1 above)
Comment (abhipsaMisra) [reply iothub/device/src/Transport/Http/HttpClientHelper.cs:239]: Unfortunately, the method signature requires options to be passed in: https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.deserializeasync?view=net-7.0.
I can make it inline. 
(This is the http layer, so our client isn't using it directly today. If we decide to revert the removal of support in the future, we can make passing in this options configurable).

File: iothub/device/src/Transport/Http/HttpClientHelper.cs [hunk 2]
```diff
@@ -236,10 +236,9 @@ private static async Task<T> ReadAsAsync<T>(HttpContent content, CancellationTok
...
             using Stream stream = await content.ReadAsStreamAsync().ConfigureAwait(false);
-            using var reader = new StreamReader(stream);
-            using var jsonReader = new JsonTextReader(reader);
-            return new JsonSerializer().Deserialize<T>(jsonReader);
+            return await JsonSerializer.DeserializeAsync<T>(stream, jsonSerializerOptions, token).ConfigureAwait(false);
```
Comment (abhipsaMisra) [inline iothub/device/src/Transport/Http/HttpClientHelper.cs:241]:
```suggestion
            return await JsonSerializer.DeserializeAsync<T>(stream, new JsonSerializerOptions(), token).ConfigureAwait(false);
```

File: iothub/device/src/Transport/Http/HttpClientHelper.cs (hunk 1 above)
Comment (abhipsaMisra) [inline iothub/device/src/Transport/Http/HttpClientHelper.cs:239]:
```suggestion
            
```

File: iothub/device/src/Transport/Http/HttpClientHelper.cs [hunk 3]
```diff
@@ -236,10 +236,9 @@ private static async Task<T> ReadAsAsync<T>(HttpContent content, CancellationTok
         {
             token.ThrowIfCancellationRequested();
 
+
```
Comment (abhipsaMisra) [inline iothub/device/src/Transport/Http/HttpClientHelper.cs:239]:
```suggestion
```

**Review by patilsnr — APPROVED** [review]

**Pull request merged**